### PR TITLE
API Summary Style Customization

### LINKED ISSUE #27: Component style extend & customization
# Component style extend & customization

## Who will benefit from this feature?

The immediate benefactors would be the Salesforce Docs team who are utilizing the ARC components to render reference documentation in their web portal, and wish to have the reference styled in a specific way.

In the long term, future consumers may also utilize the CSS variables to style it their own way if they wish.

## Overview

This is a feature request to expand the style customizations to api documentation and the sub components. My proposal is to introduce a few CSS parts and more CSS variables in a backwards compatible fashion.

List of sub-components that could see change:
- api-summary
- api-endpoint-documentation
- api-method-documentation
- api-type-documentation
- api-body-document
- api-resource-example-document
- api-type-document
- markdown-styles

## Use cases

- Style individual section titles and headings
- Increased control over spacing
- Enhanced markdown style for arc-marked markdown content

#### Do you know anyone else, internal or external, that is also interested in this feature?

Salesforce Developer WebOps team would also like to see this feature since they are consuming SF Docs.

#### Do you expect this feature to be fairly stable?  How might it be extended or changed in the future?

It is fairly stable and backwards compatible.

#### List every UI surface belonging to or potentially affected by this feature:

- Consumers of `api-documentation`
- Consumers of `markdown-styles`

#### How could this feature be abused?

Mostly styling changes so I don't expect there to be security implications.

#### Imagine you’re Dr. Evil app user, list the three worst evil deeds you could commit with your feature (if you’ve got good ones, feel free to add more):

- If a security vulnerability surfaced for CSS parts, they could leverage that somehow on the new parts introduced.
- The evil person could inject a CSS style sheet that set the style in such a way that the content is hidden from the screen.
- The evil person could make the styling look really terrible.

### PR DESCRIPTION
**Pull request by benri**
Fixes [#27](https://github.com/advanced-rest-client/api-documentation/issues/27)

- Add CSS parts to major sections
- Add api to not render the endpoints table of contents
- customize separator margin

### CODE REVIEW

File: src/Styles.js [hunk 1]
```diff
@@ -102,7 +102,7 @@ a:hover {
 .separator {
   background-color: var(--api-summary-separator-color, rgba(0, 0, 0, 0.12));
   height: 1px;
-  margin: 40px 0;
+  margin: var(--api-summary-separator-margin-y, 40px) var(--api-summary-separator-margin-x, 0);
```
Comment (jarrodek) [inline src/Styles.js]: Why not just `--api-summary-separator-margin` with a default value of `40px 0`? I don't think it's beneficial to split it into two variables.

**Comment by jarrodek** [comment]
Hey, could you please confirm CLA? @benri

File: src/Styles.js (hunk 1 above)
Comment (benri) [reply src/Styles.js]: Sure, I can do that

**Comment by benri** [comment]
@jarrodek Yes, I followed the CLA link to a form and filled out the information so that they can verify I am an internal member. Thanks!

File: src/ApiSummary.js [hunk 2]
```diff
@@ -331,8 +336,8 @@ export class ApiSummary extends AmfHelperMixin(LitElement) {
       return '';
     }
     return html`
-    <div class="api-title" role="heading" aria-level="${titleLevel}">
-    <label>API title:</label>
+    <div class="api-title" role="heading" aria-level="${titleLevel}" part="api-title">
+    <label part="api-title-label">API title:</label>
```
Comment (KrisGraySFDC) [inline src/ApiSummary.js:340]: Do we need to include the CSS Parts in the doc so people know what is capable?

Comment (benri) [reply src/ApiSummary.js:340]: That's a good idea, I'll include a doc

**Comment by KrisGraySFDC** [comment]
@jarrodek Ben did the signing of the salesforce-cla, but its obviously not picking it up.

Can we re-run the job? Is there someone I need to talk to?

**Review by twoplustwoone — CHANGES_REQUESTED** [review]
Could you bump the version so I can merge it?

**Comment by KrisGraySFDC** [comment]
@TwoPlusTwoOne is it possible to get Benri added to the github organization so his CLA will pass?
Also any way to merge this? We're hoping to get this shipped out later in the week and still need to update NPM with the latest packages, update our versions and re-style them.

**Pull request merged**